packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js

```diff
@@ -179,6 +181,126 @@ const Conversation = WebexPlugin.extend({
       });
   },
 
+  /**
+   * @private
+   * generate a deterministic HMAC for a reaction
+   * @param {Object} displayName displayName of reaction we are sending
+   * @param {Object} parent parent activity of reaction
+   * @returns {Promise<HMAC>}
+   */
+  createReactionHmac(displayName, parent) {
+    // not using webex.internal.encryption.getKey() because the JWK it returns does not have a 'k'
+    // property. we need jwk.k to correctly generate the HMAC
+
+    return this.webex.internal.encryption.unboundedStorage.get(parent.encryptionKeyUrl)
+      .then((keyString) => {
+        const key = JSON.parse(keyString);
+        // when we stringify this object, keys must be in this order to generate same HMAC as
+        // desktop clients
+        const formatjwk = {k: key.jwk.k, kid: key.jwk.kid, kty: key.jwk.kty};
+
+        const source = `${JSON.stringify(formatjwk)}${parent.id}${displayName}`;
+
+        const hmac = hex.stringify(hmacSHA256(source, parent.id));
+
+        return Promise.resolve(hmac);
+      });
+  },
+
+  /**
+   * @private
+   * send add or delete reaction to convo service
+   * @param {Object} conversation
+   * The payload to send a reaction
+   * @typedef {Object} ReactionPayload
```

Comment:
The typedef needs to be in its own jsdoc block, so insert this above line 210:

```js
/**
 * @typedef {Object} ReactionPayload
    * @property {Object} actor
   * @property {string} actor.objectType
   * @property {string} actor.id
   * @property {string} objectType
   * @property {string} verb will be either add' or 'delete'
   * @property {Object} target
   * @property {string} target.id
   * @property {string} target.objectType
   * @property {Object} object this will change on delete vs. add
   * @property {string} object.id present in delete case
   * @property {string} object.objectType 'activity' in delete case, 'reaction2' in add case
   * @property {string} object.displayName must be 'celebrate', 'heart', 'thumbsup', 'smiley', 'haha', 'confused', 'sad'
   * @property {string} object.hmac
 */
```